toolvoice/scripts/evaluation/evaluate_when2call.py

```
@@ -0,0 +1,452 @@
+    if not raw:
+        return False
+    key = re.sub(r"\s+", "_", raw.strip()).upper()
+    if key in _TOOL_CALL_LABELS:
+        return True
+    if key in _NON_TOOL_CALL_LABELS:
+        return False
+    if re.search(r"[A-Z0-9]+[._][A-Z0-9]", raw) or "." in raw:
+        return True
+    if raw.isupper() and "_" in raw:
+        return True
+    return False
```

Ruff (0.16.1)

[warning] Return the condition `bool(raw.isupper() and "_" in raw)` directly

Replace with `return bool(raw.isupper() and "_" in raw)`

(SIM103)